lib/ansible/module_utils/net_tools/netbox/netbox_utils.py

```diff
@@ -128,7 +128,10 @@ def find_ids(nb, data):
             elif k == "nat_inside":
                 if v.get("vrf"):
                     vrf_id = nb.ipam.vrfs.get(**{"name": v["vrf"]})
-                    query_id = nb_endpoint.get(**{"address": v["address"], "vrf_id": vrf_id.id})
+                    if vrf_id:
+                        query_id = nb_endpoint.get(**{"address": v["address"], "vrf_id": vrf_id.id})
+                    else:
+                        return "%s does not exist - Please create VRF" % (data["vrf"])
```

Comment:
It's also valid for the next block, but why don't you raise an exception like `ValueError`, with your message, that you catch when you call `find_ids`?

For example:

```
if vrf_id:
    query_id = nb_endpoint.get(address=v["address"], vrf_id=vrf_id.id)
else:
    raise ValueError("%s does not exist - Please create VRF" % (data["vrf"]))
```

And then in main function:

```

    try:
        if 'present' in state:
            return module.exit_json(
                **ensure_ip_address_present(nb, nb_endpoint, norm_data)
            )
        else:
            return module.exit_json(
                **ensure_ip_address_absent(nb, nb_endpoint, norm_data)
            )
    except pynetbox.RequestError as e:
        return module.fail_json(msg=json.loads(e.error))
    except ValueError as e:
        return module.fail_json(msg=str(e))
```

This way, it's more obvious than there is an error here. You can also define a custom exception for not found objects.

Reply:
The one thing I'm battling with is if we'd want it to fail closed or open. If we fail closed, that means someone can't react to the `msg` and say create the VRF the next play with a `when` condition.

What are your thoughts?

Reply:
Hmm sorry, I don't really get your question. Could you give me an example please?

Reply:
Ignore me! After thinking about this further - it makes sense to just have the playbook fail if one of the values doesn't return an id (doesn't exist)!

Reply:
How does everything look now?

I did notice we're returning a dictionary when we encounter the following which is not what we're documenting in the modules:
```
except pynetbox.RequestError as e:
        return module.fail_json(msg=json.loads(e.error))
```
If we change it to:
```
except pynetbox.RequestError as e:
        return module.fail_json(msg=e.error)
```
It returns the following:
```
TASK [Create prefix with several specified options] *********************************************************************************************************************
fatal: [localhost]: FAILED! => changed=false
  msg: '{"vlan":["Primary key must be an integer"]}'
```

Reply:
LGTM

Hmm, about the error, it's kind of tricky here… I would let it as it is, because I cannot really find a good way to do that :(

Reply:
Ok awesome! Thanks for reviewing.

I'm going to try and possibly get an `netbox_interface` and `netbox_vrf` one created within the next couple of weeks.